File: 0386-lexicographical-numbers/0386-lexicographical-numbers.cpp
class Solution {
public:
    void dfs(long long st,vector<int> &arr,int n){
        if(st > n)return;
        arr.push_back(st);
        for(int i=0;i<=9;i++){
            int nxt = (st*10LL) + i;
            if(nxt<=n)dfs(nxt,arr,n);
            else break;
        }
    }
    vector<int> lexicalOrder(int n) {
        vector<int> ans;
        for(int i=1;i<=9;i++)
            dfs(i,ans,n);
        return ans;
    }
};

File: 0386-lexicographical-numbers/0386-lexicographical-numbers.cs
public class Solution {
    public void dfs(int st,IList<int> ans,int n){
        if(st>n)return;
        ans.Add(st);
        for(int i=0;i<=9;i++){
            int nxt = st * 10 + i;
            if(nxt<=n)dfs(nxt,ans,n);
        }
    }
    public IList<int> LexicalOrder(int n) {
        IList<int> ans = new List<int>();
        for(int i=1;i<=9;i++)
            dfs(i,ans,n);
        return ans;
    }
}